src/client/blobs.rs

```diff
@@ -0,0 +1,179 @@
+use super::BlobScanClient;
+use alloy_chains::Chain;
+use serde::{ser::StdError, Deserialize, Serialize};
+use std::fmt;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlobListResponse {
+    blobs: Vec<Blob>,
+    #[serde(rename = "totalBlobs")]
+    total_blobs: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Blob {
+    #[serde(rename = "versionedHash")]
+    versioned_hash: String,
+    commitment: String,
+    proof: String,
+    size: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlobId {
+    #[serde(rename = "versionedHash")]
+    versioned_hash: String,
+    commitment: String,
+    proof: String,
+    size: u32,
+    data: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ApiError {
+    message: String,
+    code: String,
+    issues: Vec<Issue>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Issue {
+    message: String,
+}
+
+//
+// #[derive(Debug)]
+// pub struct BlobQueryParams {
+//     pub rollup: Option<String>,
+//     pub page: Option<u32>,
+//     pub page_size: Option<u32>,
+// }
+#[derive(Debug, Clone, Copy, Default)]
+pub struct BlobQueryParams {
+    pub rollup: Option<Chain>,
+    pub page: Option<u32>,
+    pub page_size: Option<u32>,
+}
+impl BlobScanClient {
+    // Retrieve  blobs.
+    pub async fn get_blobs(
+        &self,
+        params: BlobQueryParams,
+    ) -> Result<BlobListResponse, BlobScanError> {
+        let url = format!("{}/blobs", self.base_url);
+        let mut request = self.client.get(url);
+
+        
+        if let Some(rollup) = params.rollup {
+            // request = request.query(&[("rollup", rollup)]);
+            request = request.query(&[("rollup", rollup.to_string())]);
+        }
+        if let Some(page) = params.page {
+            request = request.query(&[("p", page.to_string())]);
+        }
+        if let Some(page_size) = params.page_size {
+            request = request.query(&[("ps", page_size.to_string())]);
+        }
```

Comment:
there should be a helper on the type to turn it into a `&[(k, v)]` instead so we can avoid the ifs here